crates/optimism/node/src/txpool.rs

```diff
@@ -52,9 +52,14 @@ where
         if let Ok(Some(block)) =
             this.inner.client().block_by_number_or_tag(reth_primitives::BlockNumberOrTag::Latest)
         {
-            // genesis block has no txs, so we can't extract L1 info, we set the block info to empty
-            // so that we will accept txs into the pool before the first block
-            if block.number == 0 {
+            // update L1 block info to:
+            //   * empty on two cases:
+            //     * any chain on genesis block, it has no txs, so we can't extract L1 info, by
+            //       setting L1 block info to empty we still accept tx in the pool before the first
+            //       block.
+            //     * dev chain on all blocks, it does not rely on a L1 chain.
+            //   * data coming from L1
+            if block.number == 0 || this.inner.chain_spec().chain == Chain::dev() {
```

Comment:
hmm, using the dev chain check here does not always succeed for example if a different chain id is used.

what we should do is simply always initialize the block_info with Some by duplicating this

https://github.com/paradigmxyz/reth/blob/d647624b868ac639e90a4bcd34f44a8cbe2ab428/crates/optimism/node/src/txpool.rs#L79-L81

and an unwrap_or_default I think.

we could also not use an Option here:

https://github.com/paradigmxyz/reth/blob/d647624b868ac639e90a4bcd34f44a8cbe2ab428/crates/optimism/node/src/txpool.rs#L202-L202

perhaps this would be even better, because we assume we always have l1blockinfo?

Reply:
> we could also not use an Option here:

makes sense :+1:  done here https://github.com/paradigmxyz/reth/pull/7787/commits/50a4d084c86a9bef6aefe3c2974a48cf002e5741